plugins/xdebug/xdebug.cc

```diff
@@ -409,21 +413,73 @@ XInjectResponseHeaders(TSCont /* contp */, TSEvent event, void *edata)
   return TS_EVENT_NONE;
 }
 
+static bool
+isFwdFieldValue(const char *value, int valueLen, intmax_t &fwdCnt)
+{
+  ts::TextView paramName("fwd");
+
+  if (valueLen < static_cast<int>(paramName.size())) {
+    return false;
+  }
+
+  ts::TextView val(value, valueLen);
+
+  if (ts::strcasecmp(paramName, val.prefix(paramName.size())) != 0) {
+    return false;
+  }
+
+  val.remove_prefix(paramName.size());
+
+  if (val.size() == 0) {
+    // Value is 'fwd' with no '=<count>'.
+    fwdCnt = -1;
+    return true;
+  }
+
+  const char httpSpace[] = " \t";
+
+  val.ltrim(httpSpace);
+
+  if (val[0] != '=') {
+    return false;
+  }
+
+  val.remove_prefix(1);
+  val.ltrim(httpSpace);
+
+  size_t sz  = val.size();
+  intmax_t i = ts::svtoi(val, &val);
+
+  if ((val.size() != sz) or (i < 0)) {
+    // There were crud characters after the number, or the number was negative.
+    return false;
+  }
+
+  fwdCnt = i;
+
+  return true;
+}
+
 // Scan the client request headers and determine which debug headers they
 // want in the response.
 static int
 XScanRequestHeaders(TSCont /* contp */, TSEvent event, void *edata)
 {
-  TSHttpTxn txn     = (TSHttpTxn)edata;
-  intptr_t xheaders = 0;
+  TSHttpTxn txn      = (TSHttpTxn)edata;
+  uintptr_t xheaders = 0;
+  intmax_t fwdCnt    = 0;
   TSMLoc field, next;
   TSMBuffer buffer;
   TSMLoc hdr;
 
+  // Make sure TSHttpTxnReenable(txn, TS_EVENT_HTTP_CONTINUE) is called before exiting function.
+  //
+  ts::PostScript ps(TSHttpTxnReenable, &*txn, TS_EVENT_HTTP_CONTINUE);
```

Comment:
Why the `&*txn`? Isn't `TSHttpTxn` an opaque type?

Reply:
We're supposed to pretend we don't know it's a pointer?  Why isn't of type intptr_t then?

I need this because of the problem with lvalues for class template type parameter deduction.